rusty-machine/examples/nnet-and_gate.rs

```diff
@@ -0,0 +1,89 @@
+extern crate rusty_machine;
+extern crate rand;
+
+use rand::{random, Open01};
+use std::vec::Vec;
+
+use rusty_machine::learning::nnet::{NeuralNet, BCECriterion};
+use rusty_machine::learning::toolkit::regularization::Regularization;
+use rusty_machine::learning::optim::grad_desc::StochasticGD;
+
+use rusty_machine::linalg::matrix::Matrix;
+use rusty_machine::learning::SupModel;
+
+// AND gate
+fn main() {
+    println!("AND gate learner sample:");
+
+    const THRESHOLD: f64 = 0.7;
+
+    const SAMPLES: usize = 10000;
+    println!("Generating {} training data and labels...", SAMPLES as u32);
+
+    let mut input_data = Vec::with_capacity(SAMPLES * 2);
+    let mut label_data = Vec::with_capacity(SAMPLES);
+
+    for _ in 0..SAMPLES {
+        let Open01(left) = random::<Open01<f64>>();
```

Comment:
I think perhaps some inline comments here could be helpful. If people don't read the README I can see this being confusing. (Just a very brief explanation of what we're doing).